src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -707,7 +700,7 @@ void NetworkOPsImp::processHeartbeatTimer ()
 
```

Comment:
In the above code-path we can call `setMode(omCONNECTED)` back to back: if we're in `omDISCONNECTED` (line 687) we set `omCONNECTED`, then in 698, if we're `omCONNECTED`, we again set `omCONNECTED`. 

This may be harmless (except for the extra work), but it could cause some "flapping" of the network mode (disconnected -> connected -> syncing -> connected)